ppr-ui/tests/unit/StaffPaymentDialog.spec.ts

```diff
@@ -0,0 +1,76 @@
+// Libraries
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import flushPromises from 'flush-promises'
+import { getVuexStore } from '@/store'
+import { mount, createLocalVue } from '@vue/test-utils'
+import CompositionApi from '@vue/composition-api'
+
+// Components
+import { StaffPayment as StaffPaymentComponent } from '@bcrs-shared-components/staff-payment'
+import { StaffPaymentDialog } from '@/components/dialogs'
+import { getLastEvent } from './utils'
+
+Vue.use(Vuetify)
+
+const vuetify = new Vuetify({})
+const store = getVuexStore()
+
+document.body.setAttribute('data-app', 'true')
+
+// emitted events
+const proceed = 'proceed'
+
+// Input field selectors / buttons
+const title = '.dialog-title'
+
+
+describe('Payment component', () => {
+  let wrapper: any
+  
+  beforeEach(async () => {
+    const localVue = createLocalVue()
+    localVue.use(CompositionApi)
+    localVue.use(Vuetify)
+    wrapper = mount(StaffPaymentDialog,
+      {
+        localVue,
+        store,
+        propsData: {
+          setAttach: '',
+          setDisplay: true
+        },
+        vuetify
+      })
+
+    // wait for all queries to complete
+    await flushPromises()
+  })
+
+  afterEach(() => {
+    wrapper.destroy()
+  })
+
+  it('renders with staff payment component', () => {
+    expect(wrapper.findComponent(StaffPaymentDialog).exists()).toBe(true)
+    expect(wrapper.findComponent(StaffPaymentComponent).exists()).toBe(true)
+    expect(wrapper.findAll(title).length).toBe(1)
+    expect(wrapper.find(title).text()).toBe('Staff Payment')
+    
+  })
+
+  it('Emits the cancel action', async () => {
```

Comment:
these will need to be tweaked to emit the action from the base dialog similar to the confirmation dialog tests

Reply:
done